src/nexusrpc/handler/_common.py

```diff
@@ -0,0 +1,136 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from datetime import timedelta
+from typing import (
+    Generic,
+    Mapping,
+    Optional,
+    Sequence,
+)
+
+from nexusrpc import Link, OutputT
+
+
+@dataclass(frozen=True)
+class OperationContext:
+    """Context for the execution of the requested operation method.
+
+    Includes information from the request."""
+
+    def __new__(cls, *args, **kwargs):
+        if cls is OperationContext:
+            raise TypeError(
+                "OperationContext is an abstract class and cannot be instantiated directly"
+            )
+        return super().__new__(cls)
+
+    service: str
+    """
+    The name of the service that the operation belongs to.
+    """
+
+    operation: str
+    """
+    The name of the operation.
+    """
+
+    headers: Mapping[str, str]
+    """
+    Optional header fields sent by the caller.
+    """
+
+
+@dataclass(frozen=True)
+class StartOperationContext(OperationContext):
+    """Context for the start method.
+
+    Includes information from the request."""
+
+    request_id: str
```

Comment:
FTR, an empty request ID is not rejected by the server. This string may be empty.